firmware/tools/generate_ui-about-simple.ccp.py

```diff
@@ -0,0 +1,184 @@
+"""
+
+def get_contributors(url):
+    contributors_list = []  ## Raw list of contributor names
+    logins = json.loads(requests.get(url).text)
+    try:
+        print(f"Could not reach URL - Error: {logins['status']} Messsage: {logins['message']}")
+        ## Status f.ex. 404 / message f.ex. "Not Found" or "API rate limit exceeded for...".
+        sys.exit()
+    except:
+        pass
+    
+    for gh_user in logins:
+        clean_ghuser = re.sub('[^a-zA-Z0-9-]', '.', gh_user['login']) ## replace non printable character with .
+        contributors_list.append(clean_ghuser)
+    
+    contributor_id = 0
+    contrib_format = [] ## Formated list of contributors, max 30 char per line
+    for contributor in contributors_list:
+        contributor_id +=1
+        contrib_format.append(contributor)
+        line_len = len(",".join(contrib_format).split("\n")[-1])
+        if contributor_id < len(contributors_list) and line_len+len(contributors_list[contributor_id]) >=30:
+            contrib_format.append("\n")
+
+    return(",".join(contrib_format).replace("\n,","\n")) # replace to erase first , in followup lines
+
+def generate_content(projects):
+    project_contrib = []
+    project_contrib.append("string authors_list[] = {\n")
+    project_contrib.append("    \"#   * List of contributors *  \",\n")
+    for project in projects:
+        project_contrib.append("    \" \",\n")
+        project_contrib.append(f"    \"#{project[0]}:\",\n")
+
+        url = f"https://api.github.com/repos/{project[1]}/{project[2]}/contributors?per_page={project[3]}"
+        contrib_mayhem = get_contributors(url).split("\n")
+        for line in contrib_mayhem:
+            project_contrib.append(f"    \"{line}\",\n")
+
+    project_contrib.append("    \" \"};\n")
+    return("".join(project_contrib))
+
+
+def pp_create_ui_about_simple_cpp(cpp_file, cppheader, cppcontent, cppfooter):
+    uiaboutsimplecpp_file = []
+
+    uiaboutsimplecpp_file.append(cppheader)
+    uiaboutsimplecpp_file.append(cppcontent)
+    uiaboutsimplecpp_file.append(cppfooter)
+
+    with open(cpp_file, "w", encoding="utf-8") as file:
+        file.writelines(uiaboutsimplecpp_file)
+    
+    print("Find your ui_simple_about.cpp at", cpp_file)
+
+def pp_change_ui_about_simple_cpp(cpp_file, cppcontent):
+    content = []
+    content_pattern = re.compile(r"string authors_list\[\] = {\n(?:\s+(?:.*,\n)+\s+.*};\n)", re.MULTILINE)
+
+    # Read original file
+    with open(cpp_file, 'r') as file:
+        filedata = file.read()
+
+    # Replace regex content with generated list
+    for match in content_pattern.finditer(filedata):
+            content.append(match[0])
+ 
+    filedata = filedata.replace(content[0], cppcontent)
+
+    # Write new file
+    with open(cpp_file, 'w') as file:
+        file.write(filedata)
+
+
+projects = []
+## Format: Project title, Github name, Github repo, Amount of contributors
+projects.append(["Mayhem-Firmware","portapack-mayhem","mayhem-firmware","50"])
```

Comment:
Maybe this should be "almost all" (i.e. 200) for giving kudos to people? :D

Reply:
I wanted to start small, because of the lag-issue. In the actual about page are not 200 names, even if I have not counted them myself. I have one test to make a menubar at the top, to switch between the 4 projects, in that case I would agree to give credit to all contributors, no limit needed.
Note: Because havok is archived, we also need to add a static list for deleted repos/accounts. Not implemented now.

In nay case, feel free to adjust with more users. Maybe I'll add an argument (switch to all) for that.